### ISSUE TITLE
preserve other fields during metafile download transition to torrent download

### ISSUE THREAD

**Issue by lukerg** [rakshasa/rtorrent#107]
I use RPC interface to rtorrent and have found that anything i set in the d.set_custom{1,2,3,4} fields is lost when the meta file download transitions to the actual torrent download.

I have a web front end that i use to feed magnet links to rtorrent via  RPC as well as setting the custom3 field to store the move-on-complete filesystem folder that rtorrent performs when the torrent is complete.

i have not been sucessful in compiling the rtorrent code on my archlinux box (some BS about aclocal macros not being defined) but i think the changes i want could be done similar to what is done on line 715 of file rtorrent/src/core/download_list.cc.

**Comment by larsjohnsen** [comment]
I second this request. My current work around is actively waiting for the torrent to be reloaded (see https://github.com/larsjohnsen/Sonarr/commit/255813e85565768edd65cdd894ab89af64ef0c87#diff-22bfc521b64f61bc13cbd67edccefcecR36), which is far from optimal.

**Comment by bobbintb** [comment]
I still have this issue about half the time.
https://github.com/rakshasa/rtorrent/issues/450

**Comment by pyroscope** [comment]
The linked issue was closed, not this one.

**Comment by bobbintb** [comment]
Right, I realized that after the fact and forgot to go back and edit it.

Anyway, is there anything about this? It hasn't gotten so much as an acknowledgment as a bug in the 3 and a half years since it was reported and, oddly, there are very few reports considering the aspects involved.

**Comment by pyroscope** [comment]
BTW, the place to fix this is 

https://github.com/rakshasa/rtorrent/blob/9abcea84c7f7594ef9cc99f151467cf8b718a9c0/src/core/manager.cc#L349

**Comment by bobbintb** [comment]
I found that already but I'm not a skilled enough coder to fix it myself.

**Comment by chros73** [comment]
@pyroscope : actually there is the code that supposed to acomplish this: https://github.com/rakshasa/rtorrent/blob/9abcea84c7f7594ef9cc99f151467cf8b718a9c0/src/core/manager.cc#L357

And I use it via watchdir sometimes with `load.start` and it reserves all the properties that I'm using (rtorrent-ps 0.9.6): `d.custom=data_dir` , `d.custom=meta_dir` , `d.custom=unsafe_data` , `d.priority` , `d.throttle_name`

@bobbintb try it out with pure rtorrent at first whether the variables that you want to use are indeed reversed or not. If they are then Sonarr applies something in the wrong order.

**Comment by bobbintb** [comment]
Do you have your reverse error with Sonarr or just in general? So if I understand you, you're saying you think the issue is the variables that carry over get out of order? How did you find out it was reversing them? I haven't really noticed any such thing myself, just the label missing.

One of the problems I am having though. is that it is inconsistent. It's been hard to troubleshoot because I think I fix it but it happens later on. I have not been able to find a trigger for the issue, it just happens. Usually when I am looking at it, it works fine, but that's just happenstance. Much of the time I go to check on it and there will be a torrent with no label. I recall testing it without Sonarr by repeatedly creating a magnet link with a label and it seemed to work every time but I don't know if that is because it just happened to not have a problem or if it is really Sonarr's fault. I'll try again because I am a little hazy on the specific results. But still, why would this happen only some times. I really doubt it is a Sonarr issue because all Sonarr does is send the torrent. The subsequent loosing of the label happens after that. I'll still check it out though.

**Comment by chros73** [comment]
> Do you have your reverse error with Sonarr or just in general?

No, sorry, I misspelled the "reversed" word in the last sentence, I meant "reserved" :) (Corrected now.)
I don't have any issues at all, that's what I meant. I don't use Sonarr, though. :) (And I don't use magnet links a lot with rtorrent, just sometimes.)

About the label problem of yours:
- are you sure that it only happens to magnet downloads and not with torrent files?

> I recall testing it without Sonarr by repeatedly creating a magnet link with a label 

How exactly? :) what was the exact command/config?

> or if it is really Sonarr's fault.

:D

Let me repeat myself:
> try it out with pure rtorrent at first whether the variables that you want to use are indeed reserved or not (means lost). If they are reserved then Sonarr applies something in the wrong order.

**Comment by bobbintb** [comment]
I've only had it happen with magnet torrents and with Sonarr. However, Sonarr is the only way I use magnet links. I'll have to try again because it was a while ago so I may be hazy on the specifics, but I recall trying to narrow it down to see if it was Sonarr or just magnet links. I recall I added a torrent several times through rutorrent, pasting the magnet link, giving it a label, and clicking 'add' and it worked every time. However, that doesn't tell me a whole lot since the issue is seemingly random and just may not have happened those times. I'm also not sure if that is a valid test as I'm not sure what mechanism rutorrent uses but I believe it is the same RPC call that Sonarr does. Perhaps if you could explain in greater detail what you mean by trying it with pure rtorrent and how to do so. Do you mean using some kind of tool to send an RPC?

**Comment by chros73** [comment]
> I'm also not sure if that is a valid test as I'm not sure what mechanism rutorrent uses

Me neither (since I don't use them), so you have to ask them (Sonarr as well). E.g. you refer to "label": what variable is used in rtorrent for that? :)

> trying it with pure rtorrent and how to do so. Do you mean using some kind of tool to send an RPC?

First try it without an xmlrpc util (like `pyrocore` is) just pure rtorrent:
- set up a watchdir and assign the given "label" variable for that watchdir (you have to know how "label is represented in Sonarr/rutorrent or whatever you use")
- use [this script](http://community.rutorrent.org/MagnetUri#Handling_.22magnet:.22_URIs_via_a_bash_script) to start magnets from watchdir
- check the result

Example watch dir (assigning "foo" label to torrents loaded from there):
```
schedule2 = watch_directory_1,5,5,"load.start=/path/to/watchdir/*.torrent,\"d.custom.set=label,foo\""
```

Then you can experiment later with an xmlrpc util, but you have to know for that how Sonarr is to handle things (I can't help you with that).
